apps/meteor/client/views/room/hooks/useMuteUserMutation.ts

```diff
@@ -0,0 +1,53 @@
+import { escapeHTML } from '@rocket.chat/string-helpers';
+import { useMethod, useToastMessageDispatch, useTranslation, useUserRoom } from '@rocket.chat/ui-contexts';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { roomCoordinator } from '../../../lib/rooms/roomCoordinator';
+
+export const useMuteUserMutation = (isUserMuted: boolean, rid: string) => {
+	const muteUser = useMethod(isUserMuted ? 'unmuteUserInRoom' : 'muteUserInRoom');
+	const room = useUserRoom(rid);
+	const dispatchToastMessage = useToastMessageDispatch();
+	const t = useTranslation();
+	const queryClient = useQueryClient();
+
+	const dispatchMessage = isUserMuted ? 'User__username__unmuted_in_room__roomName__' : 'User__username__muted_in_room__roomName__';
+
+	const roomName = room?.t && escapeHTML(roomCoordinator.getRoomName(room.t, room));
+
+	return useMutation(
+		['muteUser', 'unmuteUser'],
```

Comment:
I think roomId should maybe be part of the dependency array. WDYT?

Reply:
Yes, it is a good idea to include deps in query key, but in this case we don't care about the returned result nor caching it, so didn't thought of making query key unique.

Reply:
@yash-rajpal Mutation keys are optional, AFAIK. Couldn't we just omit it?

Reply:
Yep, makes no sense to have query key in such case, removed it. Thanks Tasso!